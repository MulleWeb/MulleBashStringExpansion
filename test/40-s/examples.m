#import <MulleBashStringExpansion/MulleBashStringExpansion.h>

#include <stdio.h>


static NSString  *expressions[] =
{
   @"foo.bar",
   @"url#https://",
   @"zlib%@*",
   nil,
};


int   main( void)
{
   NSString       *value;
   NSDictionary   *info;
   S              *sInstance;
   NSString       **p;

   info  = @{ @"foo": @{ @"bar": @"foobar" },
              @"bar": @"baz",
              @"url": @"https://www.github.com/archive/archive",
              @"zlib": @"zlib:mulle-c/mulle-container@*"
            };

   sInstance = [S objectWithMulleScionLocalVariables:nil
                                          dataSource:info];
   for( p = expressions; *p; p++)
   {
      value = [sInstance $:*p];
      mulle_printf( "\"%@\" -> \"%@\"\n", *p, value);
   }
   return 0;
}
