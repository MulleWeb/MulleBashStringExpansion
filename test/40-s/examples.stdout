"foo.bar" -> "foobar"
"url#https://" -> "www.github.com/archive/archive"
"zlib%@*" -> "zlib:mulle-c/mulle-container"
